### PR TITLE
Convert PeerConnection.sendOffer to async

### PR DESCRIPTION
**Pull request by dbrgn**
@lgrahl, do you have time to review?

### CODE REVIEW

File: saltyrtc/client.ts [hunk 1]
```diff
@@ -225,7 +225,14 @@ export class Client {
             }
 
             // Send offer
-            this.peerConnection.sendOffer();
+            this.peerConnection.createOffer().then(
+                (offer) => {
+                    this.peerConnection.setLocalDescription(offer); // TODO: Should we handle errors?
```
Comment (dbrgn) [inline saltyrtc/client.ts]: Not sure whether there's something we should do if setting the local description fails. Maybe errors on top level is something that should be done through event handlers?

File: saltyrtc/client.ts [hunk 2]
```diff
@@ -304,7 +307,14 @@ export class Client {
...
         if (peerConnection && !signaling && !this.keyStore.hasOtherKey()) {
-            this.peerConnection.sendOffer();
+            this.peerConnection.createOffer().then(
+                (offer) => {
+                    this.peerConnection.setLocalDescription(offer); // TODO: Should we handle errors?
+                    this._startConnectTimer();
+                    this.signaling.sendOffer(offer);
+                },
+                (error) => console.error('PeerConnection error:', error)
+            );
```
Comment (dbrgn) [inline saltyrtc/client.ts]: This code is duplicated for now.

Not sure if we should expose `createOffer` and `setLocalDescription` to the end user, or if we should go back to a `sendOffer` function.

Comment (dbrgn) [inline saltyrtc/client.ts]: OTOH if the user manages the peerconnection independently from the SaltyRTC client (like in the high level call flow), then we can remove it from the client and it doesn't matter :)

Comment (lgrahl) [inline saltyrtc/client.ts]: Exactly that.

**Comment by dbrgn** [comment]
I'll merge this for now and will separate the PeerConnection from the client in another commit.

**Pull request merged**